frontend/src/views/WatchlistsDetails.vue

```diff
@@ -6,104 +6,231 @@
-import dayjs from 'dayjs'
-import { computed, defineComponent, onBeforeMount, onUnmounted, reactive, toRefs } from 'vue'
+import { useAssetPriceStore } from '/@store/index'
+import { defineComponent, onBeforeMount, onUnmounted, reactive, ref, shallowReactive, toRef, toRefs, watch } from 'vue'
 
 
 export default defineComponent({
-  components : {AppLink, DataTable},
+  components : {AssetPriceChange, AssetSearch, AppLink, DataTable},
   props      : {
     id: {
       type     : String,
       required : true,
     },
   },
   setup (props) {
-    let liveSubscription = null
+    let liveSubscription  = null
+    let liveSubscription2 = null
+    const selectedAsset   = ref(null)
+    const watchlistId     = toRef(props, 'id')
+    const watchlistItems  = shallowReactive([])
+    const priceStore      = useAssetPriceStore()
 
     const data = reactive({
-      watchlist : null,
-      items     : [],
-      config    : {
+      watchlist   : null,
+      items       : [],
+      sortedItems : [],
+      config      : {
         fields: {
-          name   : {},
-          symbol : {},
+          name      : {},
+          ticker    : {},
+          createdAt : {
+            format: 'date'
+          },
+          dayPLChange: {
+            format : 'percent',
+            value  : (row : Holding) => {
+              if (!row.lastPrice || row.lastPrice.price === 0) {
+                return null
+              }
+
+              return (row.lastPrice.previousClose / row.lastPrice.price) - 1
+            },
+          },
         },
         headerLayout: [
-          'name',
-          'symbol',
+          [
+            'ticker', 'name'
+          ],
+          'createdAt',
+          'dayPLChange',
         ],
         settings: {
-          actions           : false,
+          actions           : true,
           translationPrefix : 'watchlist.table',
         },
       },
     })
 
-
+    async function remove (watchlistItem : WatchlistItem) {
+      watchlistItem.destroy()
+    }
+    
     onBeforeMount(async () => {
-      const q = new Query(Watchlist)
-      q.get(props.id)
+      await priceStore.subscribe()
 
+      const q          = Query.create(Watchlist)
+      data.watchlist   = await q.get(watchlistId.value)
       liveSubscription = await q.liveQuery(null, async wl => {
-        data.items     = await wl.relation('symbols').query().find()
         data.watchlist = wl
       })
+
+      const q2          = Query.create(WatchlistItem)
+      q2.equalTo('watchlist', data.watchlist)
+      q2.include('symbol')
+      liveSubscription2 = await q2.liveQuery(watchlistItems)
     })
 
     onUnmounted(async () => {
       if (liveSubscription) {
         await liveSubscription.unsubscribe()
       }
+      if (liveSubscription2) {
+        await liveSubscription2.unsubscribe()
+      }
     })
 
-    const sortedRows = computed(() => {
-      return data.items.filter(row => {
+    watch(selectedAsset, async () => {
```

Comment:
could these watchers instead be handled with computed getters and setters?
If they are required, it would be nice to have some documentation, since the code is not immediately clear

Reply:
So, from what i understood, we need to be careful with `computed`.

- `computed`  executes, and track all `ref` / `reactive` variables that were accessed.
- `watch` you can specify explicitly what to watch for, in this case, we only want to re-trigger when `selectedAsset` is modified. and NOT `data.items`

Also, when/if you have conditional logic in `computed` the behaviour will be random. in this case, if on the first execution you `return`, the rest of the reactive stuff accessed will not be registred.

`computed` is great for simple stuff, without conditional logic. Not for all situations

Reply:
So the way i understand this is that when you use a computed getter it does what you described, but a computed setter is not doing that. Since `selectedAsset` is only triggered, when an asset is selected I think using a computed setter here would be a good idea.
For `watchlistItems` it might be a different story, but Im not sure what the actual problem with a computed getter would be here. I will read a little more about it.

Reply:
I think that a computed with a setter is perfect in some situations, but here we only need to react when the `selectedAsset is updated`.

computed getter/setter are good to use when you need to do formatting for example when you have a yyyy-mm-dd input and need to sync that field to an actual Date field.

I don't really see why using `computed` everywhere instead of `watch` 😄

Reply:
From my experience using a watcher can have some unwanted side effects. In the example of the `selectedAsset` watcher, it actually changes the `selectedAsset` property, which in turn will trigger the watcher again, possibly leading to a loop.
With a computed setter you would actually store the value in another reactive property, something like `selectedAssetInternal`. So when the setter of `selectedAsset` changes `selectedAssetInternal`, the getter would update and cache the value. Which is a much more clear data flow without unwanted sideeffects.

I also find computed setter much easier to understand, since you know its meant to update the underlying property and trigger the getter. A watcher could have many different usage scenarios, which might not change a reactive value at all (like updating something in the store or posting to an api).

My understanding has been that its best to use computed properties (getter and setter) when possible.

sources:
https://vuejs.org/v2/guide/computed.html
https://v3.vuejs.org/guide/reactivity-computed-watchers.html#watching-reactive-objects
https://forum.vuejs.org/t/whats-the-difference-between-a-computed-setter-and-a-watcher/56085

Reply:
Feel free to update it if you want :), you should have write access to my fork.

Reply:
If this damn day job of mine wouldnt slow me down so much ;)

Reply:
That's why I work all night till early morning 😅 

Pro Tip: Don't !

Reply:
Could you add a todo comment to the watchers, that they should be changed to computed getter? Then we can merge this and refactor later

Reply:
Sure, I'll do that tomorow